stdlib/io/src/main/java/org/ballerinalang/stdlib/io/events/records/DelimitedRecordReadEvent.java

```diff
@@ -60,8 +63,16 @@ public DelimitedRecordReadEvent(DelimitedRecordChannel channel, EventContext con
     public EventResult get() {
         AlphaCollectionResult result;
         try {
-            String[] content = channel.read();
-            result = new AlphaCollectionResult(content, context);
+            if (channel.hasReachedEnd()) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Channel " + channel.hashCode() + " reached it's end");
```

Comment:
We need to change it's to its' I think, Please check the all other relevant places.